### PR TITLE
boards: arm: Add support for nRF51 Dongle (PCA10031)

### PR DESCRIPTION
**Pull request by stig-bjorlykke**
This commit adds support for the nrf51_pca10031 board.

Signed-off-by: Stig Bjørlykke <stig.bjorlykke@nordicsemi.no>

### CODE REVIEW

File: boards/arm/nrf51_pca10031/doc/nrf51_pca10031.rst [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+Supported Features
+==================
+
+The nrf51_pca10031 board configuration supports the following nRF51
+hardware features:
+
++-----------+------------+----------------------+
+| Interface | Controller | Driver/Component     |
++===========+============+======================+
++-----------+------------+----------------------+
```
Comment (dbkinder) [inline boards/arm/nrf51_pca10031/doc/nrf51_pca10031.rst]: delete this extra border line (it creates a blank row in the table

File: boards/arm/nrf51_pca10031/doc/nrf51_pca10031.rst [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+=========
+
+Refer to the :ref:`nordic_segger` page to learn about debugging Nordic boards with a
+Segger IC.
+
+
+Testing the LEDs on the nRF51 Dongle
+************************************
+
+There are 1 sample that allow you to test that the LED on
```
Comment (dbkinder) [inline boards/arm/nrf51_pca10031/doc/nrf51_pca10031.rst]: Let's change this whole section into:
```
Build and flash the :ref:`blinky-sample` sample to test that the onboard LED
is working properly with Zephyr.  LED definitions can be found in
:file:`boards/arm/nrf51_pca10031/board.h`.
```

**Review by dbkinder — CHANGES_REQUESTED** [review]

File: boards/arm/nrf51_pca10031/doc/nrf51_pca10031.rst (hunk 1 above)
Comment (stig-bjorlykke) [reply boards/arm/nrf51_pca10031/doc/nrf51_pca10031.rst]: This table was copied from another board's doc, I'll fix this and create a new PR to fix the table in doc for other boards.

File: boards/arm/nrf51_pca10031/doc/nrf51_pca10031.rst (hunk 2 above)
Comment (stig-bjorlykke) [reply boards/arm/nrf51_pca10031/doc/nrf51_pca10031.rst]: Actual this ```board.h``` does not exist anymore so I will remove this reference.

**Comment by jhedberg** [comment]
I've so far used the nRF51 DK board definition (nrf51_pca10028) when building for the nRF51 USB dongle, and haven't had any issues with that. What kind of improvements are to be expected with a dedicated board definition for the dongle?

**Review by ioannisg — COMMENTED** [review]
OK in general with this PR; looks good.
I am wondering, since this is so similar to nrf51_pca10028, if these two boards could be merged into the same directory. (Still separate boards, but located under same dir)

File: boards/arm/nrf51_pca10031/doc/nrf51_pca10031.rst [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
++-----------+------------+----------------------+
+| FLASH     | on-chip    | flash                |
++-----------+------------+----------------------+
+| GPIO      | on-chip    | gpio                 |
++-----------+------------+----------------------+
+| I2C(M)    | on-chip    | i2c                  |
++-----------+------------+----------------------+
+| NVIC      | on-chip    | arch/arm             |
++-----------+------------+----------------------+
+| PWM       | on-chip    | pwm                  |
```
Comment (lemrey) [inline boards/arm/nrf51_pca10031/doc/nrf51_pca10031.rst]: I reckon there is no PWM hardware on the nRF51.

Comment (stig-bjorlykke) [reply boards/arm/nrf51_pca10031/doc/nrf51_pca10031.rst]: This is a copy from nrf51_pca10028, so if this is wrong we should also update the existing board documentation.

**Comment by lemrey** [comment]
@ioannisg 
> I am wondering, since this is so similar to nrf51_pca10028, if these two boards could be merged into the same directory. (Still separate boards, but located under same dir)

Why? Each board has its own directory under `/boards` after all and those two are surely different boards.

**Review by dbkinder — APPROVED** [review]
doc changes LGTM, but see @lemrey comment about PWM support

**Comment by galak** [comment]
Any plans to update this PR or should we close it?

**Comment by ffranchina** [comment]
@stig-bjorlykke Actually I think it wouldn't be a problem to update the PR and leaving behind the docs for `nrf51_pca10028` since they are two completely different boards and their docs are independent. 

It would be possible to open a separate issue for the other board and than making another PR, I guess. (It would be also cleaner) :)

**Comment by stig-bjorlykke** [comment]
I have now removed PWM support from documentation.

**Comment by lemrey** [comment]
@anangl should have a look.

**Comment by stig-bjorlykke** [comment]
Why is the tests failing for ```QUARK_D2000``` when I didn't touch any related files?

**Comment by galak** [comment]
> Why is the tests failing for `QUARK_D2000` when I didn't touch any related files?

Since you touched `tests/drivers/adc/adc_api/src/test_adc.c` that might be causing a build on `QUARK_D2000`.  We try and be smart about what we build, but sometimes we miss things and some times we build a bit too much.

**Comment by stig-bjorlykke** [comment]
> Since you touched `tests/drivers/adc/adc_api/src/test_adc.c` that might be causing a build on `QUARK_D2000`.

Ok. It does not give any added value testing for this board so I'll update the changes.

**Comment by ioannisg** [comment]
What do we want to do with the patxh, @carlescufi? Should I merge it? @anangl could you add your review?

**Review by anangl — APPROVED** [review]
LGTM.

It would be good to update all "ok" to "okay" in the `.dts` file, for consistency with other boards. We also enable GPIO by default (in `*_defconfig` files). It would be good to do it for this board as well.

**Review by ioannisg — CHANGES_REQUESTED** [review]
I would say the suggestions by @anangl should really be done before we (decide to) merge that in. We like consistency :)

**Comment by stig-bjorlykke** [comment]
Files has been updated with requested changes and changes in Zephyr since the PR was last updated.

**Review by anangl — APPROVED** [review]

**Review by ioannisg — APPROVED** [review]

**Pull request merged**